pkg/cfn/manager/api.go

```diff
@@ -474,36 +477,35 @@ func defaultStackStatusFilter() []*string {
 
 // DeleteStackBySpec sends a request to delete the stack
 func (c *StackCollection) DeleteStackBySpec(s *Stack) (*Stack, error) {
-	for _, tag := range s.Tags {
-		if matchesClusterName(*tag.Key, *tag.Value, c.spec.Metadata.Name) {
-			input := &cloudformation.DeleteStackInput{
-				StackName: s.StackId,
-			}
-
-			if cfnRole := c.roleARN; cfnRole != "" {
-				input = input.SetRoleARN(cfnRole)
-			}
+	if !matchesCluster(c.spec.Metadata.Name, s.Tags) {
+		return nil, fmt.Errorf("cannot delete stack %q as it doesn't bear our %q, %q tags", *s.StackName,
+			fmt.Sprintf("%s:%s", api.OldClusterNameTag, c.spec.Metadata.Name),
+			fmt.Sprintf("%s:%s", api.ClusterNameTag, c.spec.Metadata.Name))
+	}
 
-			if _, err := c.cloudformationAPI.DeleteStack(input); err != nil {
-				return nil, errors.Wrapf(err, "not able to delete stack %q", *s.StackName)
-			}
-			logger.Info("will delete stack %q", *s.StackName)
-			return s, nil
-		}
+	input := &cloudformation.DeleteStackInput{
+		StackName: s.StackId,
 	}
 
-	return nil, fmt.Errorf("cannot delete stack %q as it doesn't bear our %q, %q tags", *s.StackName,
-		fmt.Sprintf("%s:%s", api.OldClusterNameTag, c.spec.Metadata.Name),
-		fmt.Sprintf("%s:%s", api.ClusterNameTag, c.spec.Metadata.Name))
-}
+	if cfnRole := c.roleARN; cfnRole != "" {
```

Comment:
I feel like creating the `cfnRole` variable is a bit of an overkill here. It's not like it's hard to read / type `c.roleARN` or looks too long or whatever. :)

Reply:
This line wasn't introduced in this PR, I moved the existing code.

Reply:
ah